packages/core/src/external/carequality/pd/validating-pd.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  Address as FHIRAddress,
+  ContactPoint,
+  Identifier,
+  Patient as FHIRPatient,
+} from "@medplum/fhirtypes";
+import { getStateEnum, USState } from "../../../domain/geographic-locations";
+import { Address } from "../../../domain/medical/address";
+import { Contact } from "../../../domain/medical/contact";
+import { PatientData, PersonalIdentifier } from "../../../domain/medical/patient";
+import { isContactType } from "../../fhir/patient";
+
+export class PatientAddressRequestedError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "PatientAddressRequestedError";
+  }
+}
```

Comment:
Please refer to how we implement `MetriportError` and its subclasses, and make sure to have these extending `MetriportError`.